### DISCUSSION TITLE
External Controller support

### DISCUSSION THREAD

**Discussion by Metal-Alvaromon** [eduardofilo/RG350_adam_image#281] [Hardware problems?]
Does anyone knows how external controller support works? I can't seem to get my controllers working on my RG350M.

**Comment by eduardofilo** [comment]
It works. You can check it on GMenu2X. But then for it to work in the different emulators, it has to be supported by themselves. I guess it will also work in RetroArch but I've never tried to configure it.

**Reply by Metal-Alvaromon** [reply]
Thanks for the reply, and also thank you for your great work! Should the controller work on the simplemenu as well? 
  I've found a version of PCSX4ALL with support for external controllers, is there a way to install it and keep my save states and save files?

**Reply by eduardofilo** [reply]
Unfortunately SimpleMenu is not one of the programs that support external controllers.
You can install that version of PCSX4All in the `apps` directory on the external card. By doing that it will automatically appear in GMenu2X in the `emulators` section. To make it usable in SimpleMenu, you will first have to add it to the configuration file `/media/data/local/home/.simplemenu/section_groups/consoles.ini` and then select it before opening the ROM by pressing Select and changing the third menu item that appears.

**Reply by Metal-Alvaromon** [reply]
I did that but now the roms don't open. I get a quick open dingux loading screen and then return to the roms selection. Here's what I did: I renamed the opk in my Mac so that it would be easier to type in the console, copied it to the external SD and then moved it to internal SD through the commander app, then I did as you said (although the last part wasn't necessary since the third menu was already set for the OPK I wanted).

**Reply by eduardofilo** [reply]
If you have not edited the `consoles.ini` file, the name of the OPK must be exactly `pcsx4all_rg350_odbeta.opk`. Is that the name you have given it?

**Reply by Metal-Alvaromon** [reply]
I discovered that the build I was trying isn't compatible with Adam Image, so that's a bummer. I'm now trying to get it working with retroarch, and this time with success! The problem now is quite ironic: once retroarch starts I can't control anything with the RG device, only with the external controller, and if the controller is not connected, I can't do anything. Any idea how I can get back to using the RG controls or both, which would be ideal?

**Reply by eduardofilo** [reply]
Take a look at the `/media/data/local/home/.retroarch/autoconfig` directory. If you find any files there, delete them. You will probably get the device controls back.
